### ISSUE TITLE
[Feature] Add option to "bdep new" for creating test/benchmark/example (sub-)packages automatically

### ISSUE THREAD

**Issue by helmesjo** [build2/build2#193]
In the [HOWTO](https://github.com/build2/HOWTO) it documents the [recommended approach](https://github.com/build2/HOWTO/blob/5a7e50c8a3eb0e92f726a5453dfca0540a7d8890/entries/handle-tests-with-extra-dependencies.md) for how to setup the project structure where test/benchmark/example packages have additional dependencies (eg. testing framework).

The end result is something like this:
```
hello/
├── ...
├── libhello
│   └── ...
├── libhello-benchmarks
│   └── ...
├── libhello-tests
│   └── ...
└── ...
```

**Suggested feature**

Given that this setup (subjectively, but still arguably) is the most common setup (anything besides trivial tests will most likely require a testing framework, same is true for benchmarks), should it be considered as an option passed to `bdep new`, eg. auto-creating the executables for listed "test" packages (tests,benchmarks,examples)? 
I guess that could be in line with the already available section which talks about [tests, examples, benchmarks](https://build2.org/bpkg/doc/build2-package-manager-manual.xhtml#manifest-package-tests-examples-benchmarks).

Something like:
`bdep new --type empty --test-packages tests,benchmarks,examples`

**Bonus:**
Related to #192, maybe an alternative would be to alter the default-generated `driver.cxx` test to emphasis that it's a smoke-test and recommended to exist (for consumers)? Maybe a comment in code, and change the file name to something self-explaining.

**Comment by Klaim** [comment]
It definitely would be useful to have one command to just setup that kind of repository/project indeed.

> `bdep new --type empty --test-packages tests,benchmarks,examples`

I would prefer (to avoid confusions) something like `bdep new -t lib,multi-packages` or something in that direction, as really we are making a library project (`-t lib`) and the fact that we want multiple packages is an "option". `-t empty` should be reserved to the case(s) where we really generate  no packages/hello-world, where `empty` makes sense.

**Comment by helmesjo** [comment]
> > `bdep new --type empty --test-packages tests,benchmarks,examples`
> 
> I would prefer (to avoid confusions) something like `bdep new -t lib,multi-packages` or something in that direction, as really we are making a library project (`-t lib`) and the fact that we want multiple packages is an "option". `-t empty` should be reserved to the case(s) where we really generate no packages/hello-world, where `empty` makes sense.

Yeah I like that one better, and indeed it makes more sense to have it on the package-creation level (instead of during project creation).
